test/e2e/tests/test_repository.py

```diff
@@ -174,6 +184,63 @@ def test_repository_lifecycle_policy(self, ecr_client):
 
         time.sleep(DELETE_WAIT_AFTER_SECONDS)
 
+        # Check ECR repository doesn't exists
+        exists = self.repository_exists(ecr_client, resource_name)
+        assert not exists
+
+    def test_repository_tags(self, ecr_client):
+        resource_name = random_suffix_name("ecr-repository", 24)
+
+        replacements = REPLACEMENT_VALUES.copy()
+        replacements["REPOSITORY_NAME"] = resource_name
+        # Load ECR CR
+        resource_data = load_ecr_resource(
+            "repository",
+            additional_replacements=replacements,
+        )
+        logging.debug(resource_data)
+
+        # Create k8s resource
+        ref = k8s.CustomResourceReference(
+            CRD_GROUP, CRD_VERSION, RESOURCE_PLURAL,
+            resource_name, namespace="default",
+        )
+        k8s.create_custom_resource(ref, resource_data)
+        cr = k8s.wait_resource_consumed_by_controller(ref)
+
+        assert cr is not None
+        assert k8s.get_resource_exists(ref)
+
+        time.sleep(CREATE_WAIT_AFTER_SECONDS)
+
+        cr = k8s.wait_resource_consumed_by_controller(ref)
+
+        # Check ECR repository exists
+        exists = self.repository_exists(ecr_client, resource_name)
+        assert exists
+
+        # Update CR
+        tags = [{
+            "key": "k1",
+            "value": "v1",
+        }]
+        cr["spec"]["tags"] = tags
```

Comment:
I think we could also benefit from another step that then updates this tag, and then another which deletes it - just to check that all code paths are working properly.